scripts/src/view/input/base.tsx

```diff
@@ -55,13 +55,16 @@ namespace fugazi.view.input {
 		public constructor(props: P, className: string, prompt?: string) {
 			super(props);
 
-			this.state.value = props.value || "";
 
 			this.prompt = prompt;
 			this.className = ["input", className];
 			this.keymap = collections.map<() => void>();
 		}
 
+		public componentWillMount() {
+			this.setState({value : this.props.value || ""});
```

Comment:
the new react typings, with the new typescript capabilities, define `this.state` and all its fields as constant. 
in addition, calling `this.setState(...)` at ctor issues a React console warning about component lifecycle.
This is the correct way (by react) to set an initial state.